src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -317,7 +323,8 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, scratchPool, layerThicknessEdge,
                flux_term3 = RediKappa(iCell) * 2.0_RKIND * &
                    (fluxRediZTop(iTr,k)/areaSum(k) - fluxRediZTop(iTr,k+1)/areaSum(k+1))
 
-               tend(iTr,k,iCell) = tend(iTr,k,iCell) + flux_term3
+               redi_term3(iTr,k,iCell) = redi_term3(iTr,k,iCell) - flux_term3
+               tend(iTr,k,iCell) = tend(iTr,k,iCell) - flux_term3
```

Comment:
This seems to be the most critical change.  It was not clear to me why flux_term3 had a positive on it and was the dominant term of 1-3 in my testing that seemed to cause the large negative temperatures.  This seems to fit best with the equation for the Redi term you had on confluence.

Reply:
Term 3 is `d/dz ( h S dot grad phi) = ( S dot grad phi)` where `grad phi` is computed on the edge and middle vertically, and then multiplied by the slope triad that goes through that edge. I don't expect a negative sign here. But just in case, I made this change in the current head of `ocean/develop` and reran this idealized SO test case.
![output](https://user-images.githubusercontent.com/4036816/74350515-0f360e00-4d73-11ea-8218-be2b2a578063.png)
Compare to identical repo with positive sign on term 3:
https://github.com/MPAS-Dev/MPAS-Model/pull/280#issuecomment-576533942
![image](https://user-images.githubusercontent.com/4036816/74350808-84094800-4d73-11ea-9b7c-3f668aa7b2b9.png)
Also, the analytic polynomial test cases match with a positive sign but not a negative sign.

Reply:
Thanks @mark-petersen, I agree the sign is correct as positive, I still don't fully understand why though.